src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -25,12 +25,35 @@ private enum ParameterSyntaxKind : byte
             DefaultParameter = 4,
         }
 
+        private class OptionalParameterValue
+        {
+            internal ConstantValue ConstantValue { get; }
+            internal BoundExpression Expression { get; }
+
+            internal OptionalParameterValue(ConstantValue constantValue) : this(constantValue, null)
+            {
+            }
+
+            internal OptionalParameterValue(ConstantValue constantValue, BoundExpression expression)
+            {
+                ConstantValue = constantValue;
+                Expression = expression;
+            }
+
+            public override string ToString()
+            {
+                if (Expression == null)
+                    return ConstantValue.ToString();
+                return $"{Expression} =({ConstantValue})";
+            }
+        }
+
         private readonly SyntaxReference _syntaxRef;
         private readonly ParameterSyntaxKind _parameterSyntaxKind;
 
         private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
         private ThreeState _lazyHasOptionalAttribute;
-        protected ConstantValue _lazyDefaultSyntaxValue;
+        private OptionalParameterValue _lazyOptionalParameterValue;
 
```

Comment:
Perhaps renaming the bound node reference to clearly indicate it is only for use in local functions and adding asserts that it is not used for other methods would be sufficient.